aux_commands/assign_group.py

```diff
@@ -0,0 +1,122 @@
+from typing import List, Dict
+
+import random
+import discord
+
+from aux_commands.create_delete_group import aux_create_group
+from aux_commands.join_leave_group import aux_join_group
+from aux_commands.open_close_groups import is_open_group
+from utils import bot_messages as btm, helper_functions as hpf
+from utils.guild_config import GUILD_CONFIG
+
+"""
+####################################################################
+#################### ASSIGN GROUP FUNCTIONS ########################
+####################################################################
+"""
+
+async def aux_assign_all(ctx):
```

Comment:
One alternative solution considering the issues you mentioned on #9 :
- Before starting adding students, we compute beforehand whether we will need new groups or not (we have slots available in current groups and also the number of students we want to assign a group).
- Then instead of using lists to gather groups of the same size, we can put them in queues. 
- Next, we follow the same group iteration process starting in line 50, and we are adding groups of size M that are receiving a new member to the queue of the groups with size M + 1
- That should solve the issue of having mismatched group sizes after the full assignation